### PR TITLE
Adding xml comments to public facing APIs in Azure.Identity

### PR DESCRIPTION
**Pull request by schaabs**

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs [hunk 1]
```diff
@@ -10,18 +10,36 @@
 
 namespace Azure.Identity
 {
+    /// <summary>
+    /// Enables authentication of a service principal in to Azure Active Directory using a X509 certificate that is assigned to it's App Registration. More information
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: >// [](start = 5, length = 2)

Should service principal be `Service Principal`?

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs [hunk 2]
```diff
@@ -10,18 +10,36 @@
...
+    /// on how to configure certificate authentication can be found here:
+    /// https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-certificate-credentials#register-your-certificate-with-azure-ad
+    /// </summary>
     public class ClientCertificateCredential : TokenCredential
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: Not related to this change, but shouldn't this be `CertificateCredentialClient` according to [https://github.com/Azure/azure-sdk/blob/master/docs/design/dotnet/DesignGuidelines.mdk#L75](https://github.com/Azure/azure-sdk/blob/master/docs/design/dotnet/DesignGuidelines.mdk#L75) ? or something like that?

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs [hunk 3]
```diff
@@ -10,18 +10,36 @@
...
         {
         }
 
+        /// <summary>
+        /// Creates an instance of the ClientCertificateCredential with the details needed to authenticate against Azure Active Directory with the specified certificate.
+        /// </summary>
+        /// <param name="tenantId">The Azure Active Directory tenant (directory) Id of the service principal.</param>
+        /// <param name="clientId">The client (application) ID of the service principal</param>
+        /// <param name="clientCertificate">The authentication X509 Certificate of the service principal</param>
+        /// <param name="options">Options that allow to configure the management of the requests sent to the Azure Active Directory service.</param>
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: Consider adding a reference to `IdentityClientOptions` so people know what the options are

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs (hunk 2 above)
Comment (schaabs) [reply sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: In general yes, types which connect to the service are suffixed with "Client".  However, for authentication, we are using the convention that all classes used to store or obtain data needed for authentication are suffixed with "Credential".  I'm creating a PR to amend the guidelines to call this out.

Comment (maririos) [reply sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: Never mind, just noticed that this is not the client itself

---
In reply to: [298278303](https://github.com/Azure/azure-sdk-for-net/pull/6731#discussion_r298278303) [](ancestors = 298278303)

File: sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs [hunk 4]
```diff
@@ -8,14 +8,27 @@
 
 namespace Azure.Identity
 {
+    /// <summary>
+    /// Provides a default <see cref="ChainedTokenCredential"/> configuration for applications that will be deployed to Azure.  The following credential
+    /// types will be tried, in order:
+    /// - <see cref="EnvironmentCredential"/>
+    /// - <see cref="ManagedIdentityCredential"/>
+    /// Consult the documentation of these credential types for more information  on how they attempt authentication.
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs]:
```suggestion
    /// Consult the documentation of these credential types for more information on how they attempt authentication.
```

File: sdk/identity/Azure.Identity/src/EnvironmentCredential.cs [hunk 5]
```diff
@@ -8,15 +8,34 @@
...
+    /// perform the authentication using these details. Please consult the
+    /// documentation of that class for more details.
+    /// </summary>
     public class EnvironmentCredential : TokenCredential
     {
         private TokenCredential _credential = null;
 
+        /// <summary>
+        /// Creates an instance of the EnvironmentCredential class and reads client secret details from environment variables.  
+        /// If the expected environment variables are not found at this time, the getToken method will return the default <see cref="AccessToken"/> when invoked.
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/EnvironmentCredential.cs]:
```suggestion
        /// If the expected environment variables are not found at this time, the GetToken method will return the default <see cref="AccessToken"/> when invoked.
```

File: sdk/identity/Azure.Identity/src/EnvironmentCredential.cs [hunk 6]
```diff
@@ -8,15 +8,34 @@
...
+        /// </summary>
         public EnvironmentCredential()
             : this(null)
         {
         }
 
+        /// <summary>
+        /// Creates an instance of the EnvironmentCredential class and reads client secret details from environment variables.  
+        /// If the expected environment variables are not found at this time, the getToken method will return the default <see cref="AccessToken"/> when invoked.
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/EnvironmentCredential.cs]:
```suggestion
        /// If the expected environment variables are not found at this time, the GetToken method will return the default <see cref="AccessToken"/> when invoked.
```

File: sdk/identity/Azure.Identity/src/EnvironmentCredential.cs [hunk 7]
```diff
@@ -29,11 +48,31 @@ public EnvironmentCredential(IdentityClientOptions options)
             }
         }
 
+        /// <summary>
+        /// Obtains a token from the Azure Active Directory service, using the specified client details specified in the environment variables 
+        /// AZURE_TENANT_ID, AZURE_CLIENT_ID, and AZURE_CLIENT_SECRET to authenticate.
+        /// </summary>
+        /// <remarks>
+        /// If the environment variables AZURE_TENANT_ID, AZURE_CLIENT_ID, and AZURE_CLIENT_SECRET are not specifeid, the default <see cref="AccessToken"/>
```
Comment (maririos) [inline sdk/identity/Azure.Identity/src/EnvironmentCredential.cs]:
```suggestion
        /// If the environment variables AZURE_TENANT_ID, AZURE_CLIENT_ID, and AZURE_CLIENT_SECRET are not specified, the default <see cref="AccessToken"/>
```

**Review by maririos — APPROVED** [review]

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs (hunk 1 above)
Comment (schaabs) [reply sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: It's not capitalized in the AAD docs so I'm just following that convention.

File: sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs (hunk 3 above)
Comment (schaabs) [reply sdk/identity/Azure.Identity/src/ClientCertificateCredential.cs]: The type will be linked in the generated docs.

**Pull request merged**